### PR TITLE
Added functions for constructing objects from SBP messages

### PR DESCRIPTION
**Pull request by jbangelo**
Added implementations of `TryFrom` for converting message types from libsbp into various types. These definitions are behind a feature flag. The following conversions are included:

- `MsgEphemerisGPS` -> `Ephemeris`
- `MsgEphemerisGal` -> `Ephemeris`
- `MsgObs` -> `Vec<NavigationMeasurement>`
- `u8` -> `Code`
- (SBP) `GnssSignal` -> `GnssSignal`
- (SBP) `GPSTime` -> `GpsTime`
- `GPSTimeSec` -> `GpsTime`

The ephemeris conversion contains conversions of `GnssSignal` and `GpsTime` so has a new error type to signal these compound error cases.

### CODE REVIEW

**Review by notoriaga — APPROVED** [review]

File: src/ephemeris.rs [hunk 1]
```diff
@@ -399,6 +399,133 @@ impl Ephemeris {
...
+            EphemerisDecodeError::InvalidSignal(other)
+        }
+    }
+}
+
+#[cfg(feature = "sbp-conversions")]
+pub use sbp_error::EphemerisDecodeError;
+
+#[cfg(feature = "sbp-conversions")]
+impl TryFrom<sbp::messages::observation::MsgEphemerisGPS> for Ephemeris {
```
Comment (notoriaga) [inline src/ephemeris.rs:442]: I think you need to either fully qualify `TryFrom` or import it, it's not in the prelude like `From`. and maybe we should add `--all-features` to `cargo check` in the jenkinsfiles

Comment (jbangelo) [reply src/ephemeris.rs:442]: Good catch! I'll add those flags to the check and clippy commands and fix up any issues.

File: src/signal.rs [hunk 2]
```diff
@@ -395,6 +395,14 @@ impl Code {
     }
 }
 
+#[cfg(feature = "sbp-conversions")]
+impl std::convert::TryFrom<u8> for Code {
```
Comment (notoriaga) [inline src/signal.rs:399]: Would it make sense to not gate this behind `sbp-conversions`? It doesn't actually use anything from `sbp`

Comment (jbangelo) [reply src/signal.rs:399]: Good point. I expect basically all of the time that we will be converting integer values into `Code` will happen when dealing with SBP messages, but I guess there could be other situations where the integer to `Code` conversions could be needed. If we do that I think we should also add a conversion for `Constellation`.

Comment (jbangelo) [reply src/signal.rs]: Changed!

File: src/signal.rs [hunk 3]
```diff
@@ -469,6 +483,17 @@ impl GnssSignal {
...
+#[cfg(feature = "sbp-conversions")]
+impl std::convert::TryFrom<sbp::messages::gnss::GnssSignal> for GnssSignal {
+    type Error = InvalidGnssSignal;
+
+    fn try_from(value: sbp::messages::gnss::GnssSignal) -> Result<GnssSignal, InvalidGnssSignal> {
+        use std::convert::TryInto;
+
+        GnssSignal::new(value.sat as u16, value.code.try_into()?)
+    }
+}
```
Comment (jbangelo) [inline src/signal.rs:495]: @notoriaga Think I should add conversions from borrowed instances as well? All of these conversions don't contain large structures that can be efficiently moved, so we don't gain anything by only consuming the input value.

**Review by silverjam — APPROVED** [review]

File: src/signal.rs (hunk 3 above)
Comment (silverjam) [reply src/signal.rs:495]: Maybe it would be possible to use `AsRef` so we don't need to duplicate the conversion functions?

Comment (jbangelo) [reply src/signal.rs:495]: Looking at the stdlib for some guidance, it seems like they only implement `From` and `TryFrom` for non-borrowed types except for types that can only by borrowed (i.e. `str` and slices). So it doesn't seem like the intended use for us to borrow with these converters.

**Pull request merged**